.github/workflows/wasi-testsuite.yml

```diff
@@ -0,0 +1,102 @@
+name: Test Wasi Testsuite
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+on:
+  push:
+    paths:
+      - ".github/workflows/wasi-testsuite.yml"
+      - "lib/host/wasi/**"
+      - "include/host/wasi/**"
+      - "thirdparty/wasi/**"
+  pull_request:
+    paths:
+      - ".github/workflows/wasi-testsuite.yml"
+      - "lib/host/wasi/**"
+      - "include/host/wasi/**"
+      - "thirdparty/wasi/**"
+
+jobs:
+  test_on_ubuntu:
```

Comment:
`test_on_ubuntu` and `test_on_macos` look very similar so I wonder if it makes sense to use [matrix strategies](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) and [if expressions](https://docs.github.com/en/actions/learn-github-actions/expressions) to remove duplications in the job definitions?

Reply:
Thanks for your advise!